ci/kind/kind-upgrade.sh

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env bash
+
+# Copyright 2023 Antrea Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -eo pipefail
+LATEST_KIND_VERSION=$(head -n2 ./ci/kind/version | tail -1)
+
+# on linux
+function upgrade_kind {
+    arch_name="$(uname -m)"
+    binary_name=""
+    case "$arch_name" in
+     "x86_64")
+      binary_name="kind-linux-amd64"
+      ;;
+      "aarch64")
+      binary_name="kind-linux-arm64"
+      ;;
+      *)
+      echoerr "Unsupported platform $arch_name"
+      exit 1
+      ;;
+    esac
+
+    set +e
+    curl -Lo ./kind https://kind.sigs.k8s.io/dl/$LATEST_KIND_VERSION/$binary_name
+    chmod +x ./kind
+    sudo mv ./kind /usr/local/bin/kind
+    if [[ $? -ne 0 ]]; then
+        echoerr "Failed to update kind version $LATEST_KIND_VERSION"
+    else
+        echo "Kind version $LATEST_KIND_VERSION updated successfully"   
+    fi
+    set -e
+}
+
+function compare_kind {
+    kind_version=$(kind version | awk '{print substr($2, 2)}') # strip leading 'v'
+    if [[ "$kind_version" != "$LATEST_KIND_VERSION" ]]; then
+        echo "=== Upgrading Kind to the latest version $LATEST_KIND_VERSION ==="
+        upgrade_kind 
+    else
+        echo "=== Existing Kind version $LATEST_KIND_VERSION is up to date ===" 
+        exit 0
+    fi
+}
+
+compare_kind
```

Comment:
I feel there is unnecessary to wrap line 50-57 into a function `compare_kind`.
It does upgrade as well after comparing, so you may just remove the function and run the check and upgrade directly in the last part of this file.